build_docs.pl

```diff
@@ -188,6 +188,23 @@ sub _find_toplevel {
     chdir $docpath;
     my $toplevel = eval { run qw(git rev-parse --show-toplevel) };
     chdir $original_pwd;
+    return $toplevel if $toplevel;
+
+    my $d = $docpath;
+    while ($d) {
+        printf "Checking for .git directory (or file) in '%s'\n", $d;
+        my $git = $d->file(".git");
+        my $git_stat = $git->stat();
+        return $git->parent if $git_stat;
+        my $p = $d->parent;
+        if ($p eq $d) {
+            # Root of this filesystem
+            $d = undef;
```

Comment:
Can we just break here? Or print an error and return immediately? Is there any other way `$d` could become undefined in this loop?